src/pivot-table/hooks/use-head-cell-dim.ts

```diff
@@ -1,16 +1,26 @@
 import type { stardust } from "@nebula.js/stardust";
+import { useOnPropsChange } from "@qlik/nebula-table-utils/lib/hooks";
 import { useState } from "react";
+import type { DataModel } from "../../types/types";
 import isEventFromColumnAdjuster from "../components/cells/utils/is-event-from-column-adjuster";
 
 interface UseHeadCellDim {
   interactions: stardust.Interactions;
+  dataModel: DataModel;
 }
 
-export const useHeadCellDim = ({ interactions }: UseHeadCellDim) => {
+export const useHeadCellDim = ({ interactions, dataModel }: UseHeadCellDim) => {
   const [open, setOpen] = useState(false);
+  const [isAdjustingWidth, setIsAdjustingWidth] = useState(false);
```

Comment:
Does it have to be a state? Can it not be a reference?

Reply:
yeah I tried with a ref and it doesn't work for selections, but it works for opening the menu... Not sure why and optimally we should probably know why. the state works at least and I can't see how this would be a big performance impact